### PR TITLE
Let C# handle mailto and external links in ReactControl (BL-10050)

### PR DESCRIPTION
**Pull request by JohnThomson**
This change is [Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/4536)

### CODE REVIEW

File: src/BloomExe/web/ReactControl.cs [hunk 1]
```diff
@@ -92,7 +92,16 @@ private void ReactControl_Load(object sender, System.EventArgs e)
...
+				
 			};
+			// This is mainly so that the mailTo: link in the Team Collection settings panel
+			// works with the user's default mail program rather than whatever GeckoFx60 does,
+			// which seems to be very buggy (BL-10050). However, I'm pretty sure there are
+			// no external links or mailTos in any of our ReactControls that we want the
+			// browser to handle itself, so for now, I'm making them all use the more reliable
+			// code in Bloom. This means, for example, that the "See how it works here" link
+			// will open the URL in the user's default browser.
+			browser.OnBrowserClick += Browser.HandleExternalLinkClick;
```
Comment (andrew-polk) [inline src/BloomExe/web/ReactControl.cs:104]: We don't need to unsubscribe?

**Review by andrew-polk — CHANGES_REQUESTED** [review]

**Review by hatton — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/4536)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @JohnThomson)

___
*[src/BloomExe/web/ReactControl.cs, line 97 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4536#-MbNzOND70PUv4xdcxGO:-MbNzOND70PUv4xdcxGP:bvmijj9) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/b9cdc6b63c997f30001cb3954ec21ded73b42a8f/src/BloomExe/web/ReactControl.cs#L97)):*
> ```Smalltalk
> 				
> 			};
> 			// This is mainly so that the mailTo: link in the Team Collection settings panel
> ```

One has to go read `HandleExernalLinkClick` to decided that this is not a bug :-) But what's a better name? Hard. Something like `InterceptIfThisIsAnExternalLinkClick`?

**Review by JohnThomson — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/4536)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @andrew-polk and @JohnThomson)

___
*[src/BloomExe/web/ReactControl.cs, line 104 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4536#-MbNjYaChvCpEAtfZSLx-r1-104:-MbaruIv2vGSEgn-WcMY:bp0shtz) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/b9cdc6b63c997f30001cb3954ec21ded73b42a8f/src/BloomExe/web/ReactControl.cs#L104)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

We don't need to unsubscribe?
</blockquote></details>

Don't see any reason to. We want this handler in place for the life of the browser. Other places where we set this up don't.

**Review by JohnThomson — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/4536)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @andrew-polk and @hatton)

___
*[src/BloomExe/web/ReactControl.cs, line 97 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4536#-MbNzOND70PUv4xdcxGO:-Mbas6Lb0vRgBe3h5qmf:b-77eo8r) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/b9cdc6b63c997f30001cb3954ec21ded73b42a8f/src/BloomExe/web/ReactControl.cs#L97)):*
<details><summary><i>Previously, hatton (John Hatton) wrote…</i></summary><blockquote>

One has to go read `HandleExernalLinkClick` to decided that this is not a bug :-) But what's a better name? Hard. Something like `InterceptIfThisIsAnExternalLinkClick`?
</blockquote></details>

I can change it if you want. But I agree, it's hard to come up with a clearly better name.

**Review by andrew-polk — APPROVED** [review]
> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/4536)* status: all files reviewed, 1 unresolved discussion (waiting on @hatton)

**Pull request merged**